### PR TITLE
ECO-624. Fix compatibility with calculation module v4.

### PR DESCRIPTION
**Pull request by ghost**
- Developer(s): @sergeysikachev

- Peer Reviewer:

- Ticket: URL_HERE

- Project PR: URL_HERE

- Academy PR: URL_HERE (for feature documentation and if needed migration guides and project integration guide)

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Bundle                | Release Type         | Constraints         |
   :--------------------- | :------------------------ | :--------------------- |
   Ratepay               |       major                 |                       |

#### Release Notes

- Fixed compatibility with calculation core module v4.
- Removed deprecated RatepayToSalesAggregatorBridge.

-----------------------------------------

#### Bundle Ratepay

_**Major:** Backwards-compatible bug fix_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated
- [ ] The change is isolated and not mixed with any cleanups or other changes
- [ ] New code fits to the architecture rules
- [ ] All new or changed business logic is covered by functional tests (in Zed business layer)

##### Change log

Bugfixes

- Fixed compatibility with calculation core module v4.
- Removed deprecated RatepayToSalesAggregatorBridge.

### CODE REVIEW

File: src/SprykerEco/Zed/Ratepay/Business/Exception/OrderTotalHydrationException.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Ratepay/Business/Exception/OrderTotalHydrationException.php]: Will we use the same license for eco as for core?

File: src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * Created by PhpStorm.
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php]: Those should not be in.

File: src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{
+
+    /**
+     * @param int $idSalesOrderItem
+     *
+     * @return \Generated\Shared\Transfer\ItemTransfer
+     */
+    public function getOrderItemTotalsByIdSalesOrderItem($idSalesOrderItem);
+
+}
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php]: This seems like the IDE is not auto-adding new lines at the end of files.

File: src/SprykerEco/Zed/Ratepay/Business/RatepayBusinessFactory.php [hunk 4]
```diff
@@ -435,6 +435,14 @@ public function createPostSaveHook()
     }
 
     /**
+     * @return \SprykerEco\Zed\Ratepay\Dependency\Facade\RatepayToCalculationInterface
+     */
+    public function getCalculation()
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Ratepay/Business/RatepayBusinessFactory.php:440]: getCalculationFacade?

File: src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php [hunk 5]
```diff
@@ -36,11 +37,31 @@ class RatepayCommunicationFactory extends AbstractCommunicationFactory
     protected $paymentMethodExtractor;
 
     /**
-     * @return \SprykerEco\Zed\Ratepay\Dependency\Facade\RatepayToSalesAggregatorInterface
+     * @return \SprykerEco\Zed\Ratepay\Dependency\Facade\RatepayToSalesInterface
      */
-    public function getSalesAggregator()
+    public function getSales()
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php]: getSalesFacade

File: src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php [hunk 6]
```diff
@@ -36,11 +37,31 @@ class RatepayCommunicationFactory extends AbstractCommunicationFactory
...
     {
-        return $this->getProvidedDependency(RatepayDependencyProvider::FACADE_SALES_AGGREGATOR);
+        return $this->getProvidedDependency(RatepayDependencyProvider::FACADE_SALES);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Ratepay\Dependency\Facade\RatepayToCalculationInterface
+     */
+    public function getCalculation()
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php]: same

File: src/SprykerEco/Zed/Ratepay/Dependency/Facade/RatepayToCalculationBridge.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+class RatepayToCalculationBridge implements RatepayToCalculationInterface
+{
+
+    /**
+     * @var \Spryker\Zed\Sales\Business\SalesFacade
+     */
+    protected $calculationFacade;
+
+    /**
+     * @param \Spryker\Zed\Sales\Business\SalesFacade $salesFacade
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Ratepay/Dependency/Facade/RatepayToCalculationBridge.php]: copy paste error?

File: src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php (hunk 5 above)
Comment (ghost) [reply src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php]: "getSalesAggregator" didn't have "Facade" part. Ok, I'll change it.

**Comment by dereuromark** [comment]
Looks like your github is not set up properly - the name and email seem to be missing.

**Comment by ghost** [comment]
@dereuromark could you please directly point to the place where you can see it?
Here is the latest commit I created. Email and name are present there.
`
commit 616a6e5a2c837fc35865e76a1ad97c7eff1c3f05
Author: Sergey Sikachev <sergey.sikachev@spryker.com>
Date:   Mon Jun 26 08:44:00 2017 +0000

    ECO-624. Update composer.lock.

`

File: src/SprykerEco/Zed/Ratepay/Dependency/Facade/RatepayToCalculationBridge.php (hunk 7 above)
Comment (ghost) [reply src/SprykerEco/Zed/Ratepay/Dependency/Facade/RatepayToCalculationBridge.php]: Yes, seems like that. My bad. Fixed.

**Comment by dereuromark** [comment]
Now it looks fine.

File: src/SprykerEco/Zed/Ratepay/Business/Exception/OrderTotalHydrationException.php (hunk 1 above)
Comment (ghost) [reply src/SprykerEco/Zed/Ratepay/Business/Exception/OrderTotalHydrationException.php]: Licence updated in all repo files.

**Comment by dereuromark** [comment]
@alex-galych We need a sniffer adjustment to auto-add the license according to https://github.com/spryker/code-sniffer/issues/68

File: src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php (hunk 2 above)
Comment (ghost) [reply src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php]: Accidentally forgotten.

File: src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php (hunk 3 above)
Comment (ghost) [reply src/SprykerEco/Zed/Ratepay/Business/Order/PartialOrderCalculatorInterface.php]: I guess, I removed it by hands. Other files created by me seem to have new lines.

File: src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php (hunk 5 above)
Comment (ghost) [reply src/SprykerEco/Zed/Ratepay/Communication/RatepayCommunicationFactory.php]: Fixed.

File: src/SprykerEco/Zed/Ratepay/Business/RatepayBusinessFactory.php (hunk 4 above)
Comment (ghost) [reply src/SprykerEco/Zed/Ratepay/Business/RatepayBusinessFactory.php:440]: Fixed.

**Pull request merged**